src/lava/magma/runtime/runtime.py

```diff
@@ -44,17 +45,15 @@ class Runtime:
     be blocking and non-blocking as specified by the run run_condition."""
 
     def __init__(self,
-                 run_cond: AbstractRunCondition,
                  exe: Executable,
                  message_infrastructure_type: ActorType):
-        self._run_cond: AbstractRunCondition = run_cond
+        self._run_cond: typing.Optional[AbstractRunCondition] = None
```

Comment:
Why is the RunCondition optional? This internal RunCondition is and was not used by the Runtime, as far as I can see. I think we do not need it, correct?